apps/authentication-service/src/shared/auth/authService.test.ts

```diff
@@ -0,0 +1,58 @@
+import fs from "fs";
+import { _testing, getAccessToken } from "./authService";
+import * as authService from "./authService";
+import { exampleToken } from "@/test/testData/users";
+
+const exampleErrorString = "Error: Authentication error";
+
+let tokenBackup: string;
+
+beforeAll(() => {
+  tokenBackup = _testing.getCachedToken() || "";
+  _testing.setCachedToken(exampleToken);
+});
+
+afterAll(() => {
+  _testing.setCachedToken(tokenBackup);
+});
+
+describe("authService", () => {
+  describe("getAccessToken", () => {
+    it("handles an error from the token endpoint", async () => {
```

Comment:
This test is 100% fake. It mocks a function to return something, and then it calls that mocked function to ensure that it returns with the thing we told it to return.

We always want to call the actual function we want to test. We mock the apis